include/umf/providers/provider_os_memory.h

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (C) 2022-2023 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+*/
+
+#ifndef UMF_OS_MEMORY_PROVIDER_H
+#define UMF_OS_MEMORY_PROVIDER_H
+
+#include "umf/memory_provider.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifdef __linux__
+#include <numaif.h>
+#include <sys/mman.h>
+#endif
+
+// TODO: expose anonymous / file-backed memory as well?
+
+// TODO: do we want this interop with linux?
+#ifdef __linux__
+enum umf_mem_protection {
+    ProtectionNone = PROT_NONE,
+    ProtectionRead = PROT_READ,
+    ProtectionWrite = PROT_WRITE,
+    ProtectionExec = PROT_EXEC
+};
+#else
+enum umf_mem_protection {
+    ProtectionNone = 0,
+    ProtectionRead = 2,
+    ProtectionWrite = 4,
+    ProtectionExec = 8
+};
+#endif
+
+#ifdef __linux__
+enum umf_mem_visibility {
+    VisibilityShared = MAP_SHARED,
+    VisibilityPrivate = MAP_PRIVATE
+};
+#else
+enum umf_mem_visibility { VisibilityShared, VisibilityPrivate };
+#endif
+
+// Corresponds to arguments to mbind
+#ifdef __linux__
+enum umf_numa_mode {
+    NumaModeDefault = MPOL_DEFAULT,
+    NumaModeBind = MPOL_BIND,
+    NumaModeInterleave = MPOL_INTERLEAVE,
+    NumaModePreferred = MPOL_PREFERRED,
+    // TODO: error 'undeclared here'
+    // NumaModeLocal = MPOL_LOCAL
+    // NumaModeStaticNodes = MPOL_F_STATIC_NODES,
```

Comment:
Because they are undeclared:
```c
./numaif.h:25:#define MPOL_DEFAULT     0
./numaif.h:26:#define MPOL_PREFERRED   1
./numaif.h:27:#define MPOL_BIND        2
./numaif.h:28:#define MPOL_INTERLEAVE  3
./numaif.h:29:#define MPOL_LOCAL       4
./numaif.h:30:#define MPOL_MAX         5
./numaif.h:33:#define MPOL_F_NODE    (1<<0)   /* return next il node or node of address */
./numaif.h:34:                          /* Warning: MPOL_F_NODE is unsupported and
./numaif.h:36:#define MPOL_F_ADDR     (1<<1)  /* look up vma using address */
./numaif.h:37:#define MPOL_F_MEMS_ALLOWED (1<<2) /* query nodes allowed in cpuset */
./numaif.h:40:#define MPOL_MF_STRICT  (1<<0)  /* Verify existing pages in the mapping */
./numaif.h:41:#define MPOL_MF_MOVE      (1<<1)  /* Move pages owned by this process to conform to mapping */
./numaif.h:42:#define MPOL_MF_MOVE_ALL (1<<2) /* Move every page to conform to mapping */
```

Reply:
Code changed